components/SSI/AddFunds/index.tsx

```diff
@@ -242,7 +243,83 @@ function Component(props: InputType) {
         }
     }
 
+    const checkBalance = async () => {
+        let addr: any = ''
+        const id = currency.toLowerCase()
+        if (originator_address?.value !== 'zilliqa') {
+            addr = originator_address?.value
+        }
+        try {
+            setLoadingInfoBal(true)
+            if (id !== 'zil') {
+                let token_addr: string
+                const init_addr = await tyron.SearchBarUtil.default.fetchAddr(
+                    net,
+                    'init',
+                    'did'
+                )
+                const get_services = await getSmartContract(
+                    init_addr,
+                    'services'
+                )
+                const services = await tyron.SmartUtil.default.intoMap(
+                    get_services.result.services
+                )
+                token_addr = services.get(id)
+                const balances = await getSmartContract(token_addr, 'balances')
```

Comment:
this code gets run twice, also in wallet info? @devwistkey

Reply:
Yes, on wallet info we need it to show balance. On this file we need it to check the balance compare to the input amount @tralkan